### PR TITLE
Fixes #19412: Switch to avocado framwork instead of testall custom script to run the tests

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/19412

### CODE REVIEW

**Comment by Fdall** [comment]
PR updated with a new commit

File: tests/quick/test_generic_methods.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/python3
+"""
+Sanity test file for ncf methods
```
Comment (amousset) [inline tests/quick/test_generic_methods.py:3]:
```suggestion
Sanity test file for methods
```

Comment (amousset) [reply tests/quick/test_generic_methods.py:3]: no big deal in tests but we should strive to remove references to "ncf" which does not make much sense

**Comment by Fdall** [comment]
PR updated with a new commit

**Review by amousset — APPROVED** [review]

**Pull request merged**